provider/keycloak_openid_user_attribute_protocol_mapper.go

```diff
@@ -0,0 +1,227 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+	"strings"
+)
+
+func resourceKeycloakOpenIdUserAttributeProtocolMapper() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKeycloakOpenIdUserAttributeProtocolMapperCreate,
+		Read:   resourceKeycloakOpenIdUserAttributeProtocolMapperRead,
+		Update: resourceKeycloakOpenIdUserAttributeProtocolMapperUpdate,
+		Delete: resourceKeycloakOpenIdUserAttributeProtocolMapperDelete,
+		Importer: &schema.ResourceImporter{
+			// import a mapper tied to a client:
+			// {{realmId}}/client/{{clientId}}/{{protocolMapperId}}
+			// or a client scope:
+			// {{realmId}}/client-scope/{{clientScopeId}}/protocolMapperId
+			State: resourceKeycloakOpenIdUserAttributeProtocolMapperImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "A human-friendly name that will appear in the Keycloak console.",
+			},
+			"realm_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The realm id where the associated client or client scope exists.",
+			},
+			"client_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "The mapper's associated client. Cannot be used at the same time as client_scope_id.",
+				ConflictsWith: []string{"client_scope_id"},
+			},
+			"client_scope_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "The mapper's associated client scope. Cannot be used at the same time as client_id.",
+				ConflictsWith: []string{"client_id"},
+			},
+			"add_to_id_token": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
```

Comment:
I recently opened #27 to use `Computed` instead of `Default` so Keycloak provides the default value instead of the provider.  It might be worth looking at some of these attributes to see if Keycloak will send back a default when it is not specified.

Reply:
I considered using `Computed` for `claim_value_type`, but Keycloak  defaults to the empty string and treats that as the `String` type rather than setting a sensible default.